agent/src/Factories/BrowserFactory.php

```diff
@@ -22,29 +22,19 @@ public function __construct(
     public function __invoke(
         float $connectionTimeout,
         float $timeout,
-        string $server,
         array $headers = [],
-        string $baseUrl = '',
-        bool $debug = false,
+        ?string $baseUrl = null,
     ): BrowserContract {
         $connector = new Connector(['timeout' => $connectionTimeout]);
 
         $browser = (new ReactBrowser($connector))
             ->withTimeout($timeout)
-            ->withHeader('nightwatch-server', $server);
-
-        if ($baseUrl) {
-            $browser = $browser->withBase($baseUrl);
-        }
```

Comment:
All headers are now applied in `agent.php` and passed into the `BrowserFactory`'s constructor.